### ISSUE TITLE
"panic: ShowContent already declared" when build project with 'gopherjs build [packagename]'

### ISSUE THREAD

**Issue by glight2000** [gopherjs/gopherjs#488]
OS : windows 10 64bit
Go : 1.6

Reproduce:
main.go
```go
package testgopherjs
import "testgopherjs/controller"
func main() {
	controller.ShowContent()
}
```
controller.go
```go
package controller
import "testgopherjs/models"
func ShowContent()  {
	models.ShowContent()
}
func ShowDetail()  {
	models.ShowContent()
}
```
contentview.go
```go
package models
//import "testgopherjs/controller"
func ShowContent()  {
	go func(){
		//controller.ShowDetail()
	}()
}
```

Use `gopherjs build [package]`, console will no response until remove the import `testgopherjs/controller` and then panic.

Here is the full stack log.
[stack.txt](https://github.com/gopherjs/gopherjs/files/358653/stack.txt)

**Comment by shurcooL** [comment]
I can reproduce what you described. I had to uncomment the 2 lines in `contentview.go` that you posted.

That Go code is invalid, there's an import cycle. The `go` command detects it right away:

```
$ go build
import cycle not allowed
package testgopherjs
	imports testgopherjs/controller
	imports testgopherjs/models
	imports testgopherjs/controller
```

But the current `gopherjs` does not:

```
testgopherjs $ gopherjs build
panic: ShowContent already declared [recovered]
	panic: ShowContent already declared

goroutine 1 [running]:
panic(0x531be0, 0xc828a86620)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
go/types.(*Checker).handleBailout(0xc8200f20e0, 0xc84befdc50)
	/usr/local/go/src/go/types/check.go:213 +0xd8
panic(0x531be0, 0xc828a86620)
	/usr/local/go/src/runtime/panic.go:443 +0x4e9
github.com/gopherjs/gopherjs/third_party/importer.(*importer).obj(0xc8429757e0, 0xc82a9e21e0)
	/Users/Dmitri/Dropbox/Work/2013/GoLand/src/github.com/gopherjs/gopherjs/third_party/importer/import.go:140 +0x3ae
github.com/gopherjs/gopherjs/third_party/importer.ImportData(0xc8200f4f90, 0xc8388cf320, 0x32, 0x32, 0x0, 0x9, 0x0, 0x0)
	/Users/Dmitri/Dropbox/Work/2013/GoLand/src/github.com/gopherjs/gopherjs/third_party/importer/import.go:71 +0x5c5
github.com/gopherjs/gopherjs/compiler.ReadArchive(0xc82ae4da40, 0x4a, 0xc82ddcb9c1, 0x13, 0xb62178, 0xc82002d0f8, 0xc8200f4f90, 0x0, 0x0, 0x0)
	/Users/Dmitri/Dropbox/Work/2013/GoLand/src/github.com/gopherjs/gopherjs/compiler/compiler.go:242 +0x3d6
github.com/gopherjs/gopherjs/build.(*Session).BuildPackage(0xc8200f6380, 0xc83a7322d0, 0x0, 0x0, 0x0)
	/Users/Dmitri/Dropbox/Work/2013/GoLand/src/github.com/gopherjs/gopherjs/build/build.go:550 +0xdbf
github.com/gopherjs/gopherjs/build.(*Session).buildImportPathWithSrcDir(0xc8200f6380, 0xc82ddcb9c1, 0x13, 0xc8288bd770, 0x42, 0x0, 0xc842976088, 0x0, 0x0)
	/Users/Dmitri/Dropbox/Work/2013/GoLand/src/github.com/gopherjs/gopherjs/build/build.go:472 +0x1a2
github.com/gopherjs/gopherjs/build.(*Session).BuildPackage(0xc8200f6380, 0xc827193db0, 0x0, 0x0, 0x0)
...
```

Luckily, this is an issue that will be resolved by #485:

```
$ gopherjs build
import cycle not allowed
package testgopherjs
	imports testgopherjs/controller
	imports testgopherjs/models
	imports testgopherjs/controller
```

**Comment by shurcooL** [comment]
I'll close this issue because it's a duplicate of #414.
